x-pack/plugins/grokdebugger/public/sections/grokdebugger/components/event_output/event_output.js

```diff
@@ -4,37 +4,29 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { uiModules } from 'ui/modules';
-import template from './event_output.html';
-import './event_output.less';
-import 'ace';
+import React from 'react';
+import {
+  EuiFormRow,
+  EuiCodeEditor
+} from '@elastic/eui';
 
-const app = uiModules.get('xpack/grokdebugger');
-
-app.directive('eventOutput', function () {
-  return {
-    restrict: 'E',
-    template: template,
-    scope: {
-      structuredEvent: '='
-    },
-    bindToController: true,
-    controllerAs: 'eventOutput',
-    controller: class EventOutputController {
-      constructor($scope) {
-        $scope.aceLoaded = (editor) => {
-          this.editor = editor;
-          editor.getSession().setUseWrapMode(true);
-          editor.setOptions({
-            readOnly: true,
-            highlightActiveLine: false,
-            highlightGutterLine: false,
-            minLines: 20,
-            maxLines: 25
-          });
-          editor.$blockScrolling = Infinity;
-        };
-      }
-    }
-  };
-});
+export function EventOutput({ value }) {
+  return (
+    <EuiFormRow
+      label="Structured Data"
+      data-test-subj="aceEventOutput"
+    >
+      <EuiCodeEditor
```

Comment:
How about wrapping this in `EuiPanel`? Same for PatternInput, EventInput, and the code editor in CustomPatternsInput.

```jsx
      <EuiPanel paddingSize="s">
        <EuiCodeEditor
          mode="json"
          isReadOnly
          width="100%"
          height="340px"
          value={JSON.stringify(value, null, 2)}
          setOptions={{
            highlightActiveLine: false,
            highlightGutterLine: false,
          }}
        />
      </EuiPanel>
```

Then it will look like this:

![image](https://user-images.githubusercontent.com/1238659/41755926-82b94dba-758e-11e8-8a7d-c2a8b47443dd.png)